### PR TITLE
[kube-prometheus-stack] Add option to create aggregate ClusterRoles

### PR DESCRIPTION
**Pull request by Wikiwix**
Signed-off-by: Philip Sieder <philip.sieder@tallence.com>

#### What this PR does / why we need it:

The PR adds a new option to create aggregate Roles which allow RBAC based on the built in ClusterRoles `view` `edit` and `admin` as described for the prometheus-operator [here](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/rbac-crd.md).

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 34.1.1
+version: 34.2.0
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:26]:
```suggestion
version: 34.4.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by Wikiwix** [comment]
@monotek GitHub's UI confuses me. It tells me that there are still changes requested.

But as far as I can see you already performed the necessary changes?

**Review by monotek — CHANGES_REQUESTED** [review]
Seems to me the labels are not added to the aggreagted rules so deinstalling the chart does not remove the admin, edit & view clusterrorles.

Can you please check / fix?

**Comment by Wikiwix** [comment]
> Seems to me the labels are not added to the aggreagted rules so deinstalling the chart does not remove the admin, edit & view clusterrorles.
> 
> Can you please check / fix?

The aggregated roles (`view, `edit` and `admin`) are not part of this chart at all ([but predefined](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles), so why should or would those be deleted by a `helm uninstall`?

If these ClusterRoles (`view, `edit` and `admin`) are *not* present, the `ClusterRoles` we create (by this addition) will not alter any other resources.

But perhaps I misunderstand you?

**Comment by monotek** [comment]
Sorry, my fault. You're right.
The role which should be deleted are kps-kube-prometheus-stack-prometheus-crd-view and so on and they are removed....

**Review by monotek — APPROVED** [review]

**Pull request merged**